explorer/explorer.go

```diff
@@ -226,6 +226,7 @@ func (exp *explorerUI) Store(blockData *blockdata.BlockData, _ *wire.MsgBlock) e
 		IdxBlockInWindow:  blockData.IdxBlockInWindow,
 		IdxInRewardWindow: int(newBlockData.Height % exp.ChainParams.SubsidyReductionInterval),
 		DevAddress:        exp.ExtraInfo.DevAddress,
+		DevFund:           exp.ExtraInfo.DevFund,
```

Comment:
But how is `exp.ExtraInfo.DevFund` initially set?

Reply:
line 175 initializes the value to 0 when explorerUI is initialized.  Since startup values of 0 until the first block should not be a problem I figured this was ok.

https://github.com/decred/dcrdata/blob/1bb5fb11b2efa701639666e41b7678e315c3b656/explorer/explorer.go#L170-L180

Reply:
That's what I'm going after.  How about updating exp.ExtraInfo instead of allocating a new one?  That way the update dev fund go routine will not lose it's changes.

Reply:
I haven't tried this, but it strikes me that reallocating it might be party of the issue...

Reply:
@chappjc good idea.  I can try that!

Also, regarding thread safe I've been trying to understand thread safe in go.  In the current design the mutex locks it up before updating and unlocks it after dev fund is kicked off which also locks and unlocks. - therefore I *think* its thread safe.    Having said that, Its not clear to me how those two routines would  not occasionally overlap locks or try to unlock something that was already unlocked.

Reply:
you are right, that mutex serves the purpose.

Reply:
updating ExtraInfo rather than a new instance seems to work great.  I can update this PR so you can test as well.

Also, what would you think about changing this:

```go
	if !exp.liteMode {
		go exp.updateDevFundBalance()
	}

	exp.NewBlockDataMtx.Unlock()
```

to this:

```go
        exp.NewBlockDataMtx.Unlock()

	if !exp.liteMode {
		go exp.updateDevFundBalance()
	}
```

Seems cleaner and insures the lock is released before we try to relock for the dev fund update.

Reply:
Great, please update with the update so I can have a look.

The suggested lock change is probably fine, but note that since `updateDevFundBalance` runs as a goroutine there's no deadlock.  Rather, if the machine has enough cores, `updateDevFundBalance` will wait on the lock, and run as soon as `Store` unlocks.  If instead we unlock first, there's likely going to be a longer window where the lock is free and when `updateDevFundBalance` gets it.  Anyway, if this were really a requirement, I would have made a comment and explained why, but I didn't, so it's no big deal.  I just wanted to let you know the thinking behind the current ordering.